iothub/service/src/JobClient/JobClient.cs

```diff
@@ -31,6 +31,10 @@ public static JobClient CreateFromConnectionString(string connectionString)
         /// <returns> A JobClient instance. </returns>
         public static JobClient CreateFromConnectionString(string connectionString, HttpTransportSettings transportSettings)
         {
+            if (transportSettings == null)
```

Comment:
this is fine since the caller at line 21 creates a new transport setting before calling into this method.